apex/contrib/csrc/groupbn/nhwc_batch_norm_kernel.h

```diff
@@ -1289,23 +1533,27 @@ __global__ __launch_bounds__(THREADS_PER_CTA, DESIRED_OCCUPANCY)
                     float x1_math[ELEMENTS_PER_LDG];
                     ldg_stream(x1_math, &gmem_src1[(is_valid ? idx : 0)*params.c]);
                     add(x_math, x1_math);
-                    unsigned int relu_mask;
+                    bitmask_t relu_mask;
+#ifdef __HIP_PLATFORM_HCC__
+                    int lane_id = threadIdx.x & 63;
+#else
                     int lane_id = threadIdx.x & 31;
+#endif
                     #pragma unroll
-                    for (int i = 0; i < ELEMENTS_PER_LDG; ++i) {
-                        bool rectified = x_math[i] < 0.0F;
-                        unsigned int local_relu_mask = __ballot_sync(0xFFFFFFFFU, rectified);
-                        if (lane_id == i) {
+                    for (int j = 0; j < ELEMENTS_PER_LDG; ++j) {
+                        bool rectified = __hle(__float2half(x_math[j]), zero_h);
+                        bitmask_t local_relu_mask = ballot(rectified);
+                        if (lane_id == j) {
```

Comment:
why not make this section of code an ifdef so that the path if preserved in cuda?

Reply:
Yes, that's a good idea.  I will fix this.

Reply:
I have fixed this